build.gradle.kts

```diff
@@ -114,6 +115,10 @@ allprojects {
         mavenCentral()
     }
 
+    project.ext {
+        this["platform"] = properties["platform"] ?: currentPlatform
+    }
```

Comment:
This could be:
```kotlin
val platform : String by project.extra {
    project.properties["platform"].toString() ?: currentPlatform
}
```
(You may be able to just do `by extra` without the `project`, idk, try it).

That would allow your code below to be simplified because `platform` would be in scope:
```kotlin
tasks.withType<ShadowJar> {
    classifier = platform
}
```

Reply:
This sets a property on the project so the subprojects know what platform to get artifacts for.
It also seems to be the only way to get the script to compile without an exception:
```
FAILURE: Build failed with an exception.
* What went wrong:
A problem occurred configuring root project 'shuffleboard'.
> Cannot configure the 'publishing' extension after it has been accessed.
```

Reply:
Ah, interesting, then `val platform  : String by project.ext` should work exactly the same.
If it doesn't something's wrong.

Reply:
Work where? That's a property use, not a property assignment. It also wouldn't be in scope for the shadowjar task anyway, since that's in a separate project configuration block.

Reply:
Oh, I opened it up and it looked like it was in the same block. My bad.